app/main.py

```diff
@@ -1,34 +1,71 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if start[0] == end[0]:
+            self.decks = [
+                Deck(start[0], column)
+                for column in range(start[1], end[1] + 1)
+            ]
+        elif start[1] == end[1]:
+            self.decks = [
+                Deck(row, start[1])
+                for row in range(start[0], end[0] + 1)
+            ]
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(
+            self,
+            row: int,
+            column: int
+    ) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(
+            self,
+            row: int,
+            column: int
+    ) -> str:
+        deck = self.get_deck(row, column)
+        deck.is_alive = False
+        if not any(deck.is_alive for deck in self.decks):
+            self.is_drowned = True
+            return "Sunk!"
+        else:
+            return "Hit!"
```

Comment:
```suggestion
            return "Sunk!"
        return "Hit!"
```
`else` redundant in this case